api/v1alpha1/trunk_type.go

```diff
@@ -0,0 +1,158 @@
+/*
+Copyright 2025 The ORC Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+// TrunkFilter specifies a filter to select a trunk. At least one parameter must be specified.
+// +kubebuilder:validation:MinProperties:=1
+type TrunkFilter struct {
+	// name of the existing resource
+	// +optional
+	Name *OpenStackName `json:"name,omitempty"`
+
+	// description of the existing resource
+	// +optional
+	Description *NeutronDescription `json:"description,omitempty"`
+
+	// portRef is a reference to the ORC Port which this trunk is associated with.
+	// +optional
+	PortRef *KubernetesNameRef `json:"portRef,omitempty"`
+
+	// projectRef is a reference to the ORC Project this resource is associated with.
+	// Typically, only used by admin.
+	// +optional
+	ProjectRef *KubernetesNameRef `json:"projectRef,omitempty"`
+
+	FilterByNeutronTags `json:",inline"`
+}
+
+// Subport represents a subport that will be attached to a trunk.
+type Subport struct {
+	// portRef is a reference to the ORC Port which will be used as a subport.
+	// +required
+	PortRef KubernetesNameRef `json:"portRef"`
+
+	// segmentationType is the type of segmentation to use (e.g., "vlan").
+	// +required
+	// +kubebuilder:validation:MaxLength=64
+	SegmentationType string `json:"segmentationType"`
+
+	// segmentationID is the segmentation identifier (e.g., VLAN ID).
+	// +required
```

Comment:
We might want to add extra validation here (if the segmentation type is vlan, we should check the value is between 1 and 4094).